bigframes/core/compile/scalar_op_compiler.py

```diff
@@ -794,14 +795,27 @@ def isin_op_impl(x: ibis_types.Value, op: ops.IsInOp):
 @scalar_op_compiler.register_unary_op(ops.ToDatetimeOp, pass_op=True)
 def to_datetime_op_impl(x: ibis_types.Value, op: ops.ToDatetimeOp):
```

Comment:
I do worry this operation is getting a bit too complicated. Do the type rules reflect the fact that a DateTime will be returned for `utc==False`?.

Reply:
Will work on the refactor in a separate PR :-)